### PR TITLE
Revamp voice version handling and cache cleanup

### PR DESCRIPTION
**Pull request by lumpidu**
This is a bug fix for deleting cached utterances in case of fast voices. Because of the calculation of the RTF (realtime factor) even for the case of reading from cached utterances, any cache item was deleted after being played, if the setting `rm_cache_item_for_fast_voices` was set to true, which is the default.
Furthermore, the missing version info for the cache item caused cache handling bugs that took the version info into account.

- don't delete cache items in case it's already inside the cache
- add sanity check for RTF-calculations > 500, which are a hint on either cancellations or problems and don't delete the cache item in that case
- if the time measured between enquing a processing request and the processing response is 0, don't calculate RTF as infinity
- always add the version of the voice for any cache item
- add dummy version info v0 for network voices
- migrate any existing network voice version in the db to v0
- assert on missing version info
- add new db migrations 6->7 for the voices table
- fix missing migration tests

### CODE REVIEW

File: app/src/main/java/com/grammatek/simaromur/AppRepository.java [hunk 1]
```diff
@@ -362,11 +361,11 @@ public void startNetworkSpeak(String voiceId, CacheItem item, String langCode,
      * @param ttsRequest       the TTSRequest for the current request
      * @return true in case audio speech entry has been found and playback started, false otherwise
      */
-    private boolean playIfAudioCacheHit(String voiceId, CacheItem item, TTSAudioControl.AudioFinishedObserver finishedObserver, TTSRequest ttsRequest) {
+    private boolean playIfAudioCacheHit(String voiceId, String voiceVersion, CacheItem item, TTSAudioControl.AudioFinishedObserver finishedObserver, TTSRequest ttsRequest) {
         UtteranceCacheManager ucm = App.getAppRepository().getUtteranceCache();
         // TODO: voiceVersion parameter is not taken into account yet !
```
Comment (bnika) [inline app/src/main/java/com/grammatek/simaromur/AppRepository.java:366]: is this TODO still valid?

File: app/src/main/java/com/grammatek/simaromur/AppRepository.java [hunk 2]
```diff
@@ -390,12 +389,12 @@ private boolean playIfAudioCacheHit(String voiceId, CacheItem item, TTSAudioCont
      * @param ttsRequest  the TTSRequest to be passed to the TTSObserver
      * @return true in case audio speech entry has been found and playback started, false otherwise
      */
-    private boolean playIfAudioCacheHit(String voiceId, CacheItem item, TTSObserver ttsObserver, TTSRequest ttsRequest) {
+    private boolean playIfAudioCacheHit(String voiceId, String voiceVersion, CacheItem item, TTSObserver ttsObserver, TTSRequest ttsRequest) {
         Log.v(LOG_TAG, "playIfAudioCacheHit()");
         UtteranceCacheManager ucm = App.getAppRepository().getUtteranceCache();
         // TODO: voiceVersion parameter is not taken into account yet !
         final List<byte[]> audioBuffers =
```
Comment (bnika) [inline app/src/main/java/com/grammatek/simaromur/AppRepository.java:396]: same - isn't the voiceVersion already added?

File: app/src/main/java/com/grammatek/simaromur/device/TTSEngineController.java [hunk 3]
```diff
@@ -248,7 +248,7 @@ public void run() {
             UtteranceCacheManager ucm =  App.getAppRepository().getUtteranceCache();
             // TODO: voiceVersion parameter is not taken into account yet !
             final List<byte[]> audioBuffers =
```
Comment (bnika) [inline app/src/main/java/com/grammatek/simaromur/device/TTSEngineController.java:250]: same - TODO comment

**Review by bnika — APPROVED** [review]

File: app/src/main/java/com/grammatek/simaromur/AppRepository.java (hunk 1 above)
Comment (lumpidu) [reply app/src/main/java/com/grammatek/simaromur/AppRepository.java:366]: nope, will remove

File: app/src/main/java/com/grammatek/simaromur/AppRepository.java (hunk 2 above)
Comment (lumpidu) [reply app/src/main/java/com/grammatek/simaromur/AppRepository.java:396]: jeps

File: app/src/main/java/com/grammatek/simaromur/device/TTSEngineController.java (hunk 3 above)
Comment (lumpidu) [reply app/src/main/java/com/grammatek/simaromur/device/TTSEngineController.java:250]: jo

File: app/src/main/java/com/grammatek/simaromur/AppRepository.java (hunk 2 above)
Comment (lumpidu) [reply app/src/main/java/com/grammatek/simaromur/AppRepository.java:394]: removed

**Pull request merged**